src/engine/QueryPlanner.cpp

```diff
@@ -672,24 +722,25 @@ vector<QueryPlanner::SubtreePlan> QueryPlanner::getOrderByRow(
 // _____________________________________________________________________________
 void QueryPlanner::getVarTripleMap(
     const ParsedQuery& pq,
-    ad_utility::HashMap<string, vector<SparqlTriple>>& varToTrip,
-    ad_utility::HashSet<string>& contextVars) const {
+    ad_utility::HashMap<string, vector<SparqlTriple>>* varToTrip,
+    ad_utility::HashSet<string>* contextVars) const {
   for (auto& t : pq._rootGraphPattern._whereClauseTriples) {
     if (isVariable(t._s)) {
-      varToTrip[t._s].push_back(t);
+      (*varToTrip)[t._s].push_back(t);
     }
     if (isVariable(t._p)) {
-      varToTrip[t._p].push_back(t);
+      (*varToTrip)[t._p._iri].push_back(t);
     }
     if (isVariable(t._o)) {
-      varToTrip[t._o].push_back(t);
+      (*varToTrip)[t._o].push_back(t);
     }
     // TODO: Could use more refactoring.
     // In Earlier versions there were no ql:contains... predicates but
     // a symmetric <in-text> predicate. Therefore some parts are still more
     // complex than need be.
-    if (t._p == CONTAINS_WORD_PREDICATE || t._p == CONTAINS_ENTITY_PREDICATE) {
-      contextVars.insert(t._s);
+    if (t._p._iri == CONTAINS_WORD_PREDICATE ||
```

Comment:
Is the above TODO still valid? This doesn't look too complicated

Reply:
I'm not sure. Do we know who added this TODO?

Reply:
According to `git blame` it was added by Björn 2 years ago, so I would say that's a no